go/README.md

```diff
@@ -1,9 +1,43 @@
-This directory contains an example usage of the GraphQL parser and AST
-library from Go. See [cgo's
-documentation](https://github.com/golang/go/wiki/cgo), particularly
-the "Function pointer callbacks" section, for explanation of the
-mechanisms in use.
-
-To build, first build the main GraphQLParser library in the parent
-directory, and then set this directory to be your `GOPATH` and run `go
-build`.
+# About
+This directory contains an example using the libgraphqlparser C library from [Go](https://golang.org/project/).
+
+For an overview of binding to C libraries in Go, please see the [cgo documentation](https://github.com/golang/go/wiki/cgo).
+Specifically, please read the overview of [Function pointer callbacks](https://github.com/golang/go/wiki/cgo#function-pointer-callbacks) in Go and C.
+
+To build with Go, please ensure that you have `pkg-config` installed for your
+system:
+
+## Installing pkg-config
+
+### Installing pkg-config on OSX
+pkg-config unfortunately does not come pre-packaged with OSX. Please install
+pkg-config via Homebrew or MacPorts:
+```sh
+# via Homebrew
+sudo brew install pkg-config
```

Comment:
typical usage for homebrew is not to use sudo. Perhaps it might be best to leave the examples out and let people use Google if they need further help with developer tools.